jpyinterpreter/src/main/java/org/optaplanner/jpyinterpreter/implementors/StackManipulationImplementor.java

```diff
@@ -175,6 +177,54 @@ public static StackMetadata shiftTOSDownBy(FunctionMetadata functionMetadata, St
         return out;
     }
 
+    /**
+     * Swaps TOS with TOS[posFromTOS]
+     *
+     * (i.e. ..., TOS[posFromTOS], ..., TOS2, TOS1, TOS -> ..., TOS, ..., TOS2, TOS1, TOS[posFromTOS])
+     */
```

Comment:
Or maybe there needs to be a `README` or a package-level Javadoc explaining all the jargon used throughout the module? A high-level description of how the code works and what it does would be useful anyway.

Reply:
I'll be writing a developer guide later. The term "TOS" means "Top of Stack", and has to do with the fact both Java and CPython use a stack-based virtual machine. Similarly, TOS1 means "1 below top of stack", TOS[x] means "x below top of stack", etc.